src/main/java/net/dancier/dancer/authentication/service/AuthenticationService.java

```diff
@@ -205,7 +205,7 @@ public void checkPasswortCodeRequestAndCreateNew(String code, String newPassword
         PasswordResetCode passwordResetCode = this.passwordResetCodeRepository
                 .findByCode(code)
                 .orElseThrow(
-                        () -> new BusinessException("No such code"));
+                        () -> new AppliationException("No such code"));
```

Comment:
I changes this exception type so that the frontend can show a proper error message like "Der code existiert nicht oder ist abgelaufen, bitte fordere einen neuen an."

Reply:
Ok, we have not talked about that yet:
I introduced an Exception-hierachy on the top with two Exceptions

BusinessExceptions and ApplicationExecptions.
Mabe the wording ist not good. What I wanted to achieve:

Differentiate between technical and functional failures. Because the would be handled fundamentally different:

technical failures I would expect the dev-team of the service to figure out how to solve the problem.
functional faiures ( I would call them business failure) are of the following kind: the client requests us to deliver a resource that is not existing. So we return a 404 which is no failure on our side. Maybe on the side of the client. Maybe even there it is correct.
Also if the client used an expired password reset code, this would be more of a business failure: our system acts as expected. It refuses to reset the password.
So I considered ApplicationException as those that shows a problem (technically) in our system.
While the rest are Business Exceptions.
OK? Understood? See a benefit in this?

Reply:
But then using an applicationException here would be correct on some cases, because the user could just have used a wrong link. And he can fix it on its own by requesting a new link. If you think different and want to stay with the business exception we would need to refactor it as it at the moment returns a 500 internal server error to the frontend without a message, so the frontend cannot handle it in the correct way... is it a gateway timeout and the user should try again later or is the code wrong and the user should request a new one. I would also prefer to return a 4xx to the frontend as for me its a client error. The code used in the request payload is wrong or outdated, even if the code was coming from the service via email.

Reply:
So you would say:
let's treat it as a client error (which would be from my point of view a Business Exception) and return a 400.
So we would add in the AuthenticationController an ExeceptionHandler and return a 400 appropriately?
From a security perspective, I would not be sure what Johannes Hoff. would say if we return concert information about the problem. Something like "code is outdated" or "code does not exist. For maybe the same reasoning, as not returning for a password login if the password or the username was the problem.
So maybe just say:
- Unable to process process validation code?

Reply:
we do not need to return information if we the code is outdated or does not exist, only the information that the code is not usable, then we could show an error message so that the user can request a new one. Otherwise we cannot distinguish between general server problems where the user should just try it again later or where the code is not usable and he should request a new one. And for me just saying that something is wrong with the code is no security leak.
Yes, it would be a BusinessException then, I mixed them up, and for me a 4xx and not a 5xx.

Reply:
OK, thats what I thought. Do you change it?

Reply:
I'll do it